### DISCUSSION TITLE
Disconnection UX (e.g. "Server is full" instead of "Connection Rejected").

### DISCUSSION THREAD

**Discussion by maxgmer** [FirstGearGames/FishNet#610] [Feature Request]
Would be awesome if clients had more info about the reason they got disconnected, as it's weird to show "Connection rejected" for any disconnect reason even though we know the reasons under the hood.

**Comment by FirstGearGames** [comment]
This can be done by using a custom authenticator and sending a disconnect reason. The PasswordAuthenticator example shows how to do this.

**Reply by maxgmer** [reply]
So when I set "Maximum Clients" to 1000, for example, in reality "Maximum Clients" will be 990, because I need to leave a gap so that connections reach authenticator, right?
